### ISSUE TITLE
Snowplow.createTracker takes 10 seconds?

### ISSUE THREAD

**Issue by peterdk** [snowplow/snowplow-android-tracker#620]
**Describe the bug**
When initiating Snowplow it takes 10 seconds to complete that call.....

**To Reproduce**
 val network = NetworkConfiguration(endpoint = "url", HttpMethod.POST)
 val config = TrackerConfiguration(context.packageName).screenViewAutotracking(true)
Snowplow.createTracker(
                context = context,
                namespace = "namespace",
                network = network,
                config,
            )

**Expected behavior**
200ms setup time or something reasonable.

**Device information (please complete the following information):**
 - Device: Pixel 6a
 - OS: Android 14
 - Version com.snowplowanalytics:snowplow-android-tracker:5.2.0

**Additional context**
Not useable this way

**Comment by peterdk** [comment]
Also on Android 9, Samsung Note

**Comment by matus-tomlein** [comment]
Hi @peterdk, that is very unexpected, could you please help me replicate this?

I have created a blank new Android app and pasted in the code that you provided above, [see here](https://github.com/matus-tomlein/snowplow-android-demo-test-lag/blob/main/app/src/main/java/com/example/testlagapp/MainActivity.kt#L42-L51).
However, when I try to test, the `createTracker` call is instant:

```
D/Snowplow: Fri Jun 23 12:58:45 GMT+02:00 2023 Snowplow tracker starting
I/TetheringManager: registerTetheringEventCallback:com.example.testlagapp
D/Snowplow: Fri Jun 23 12:58:45 GMT+02:00 2023 Snowplow tracker created
```

Could you please tell me what is different about your setup or help me extend the app to replicate the problem?

**Comment by peterdk** [comment]
Yeah, I just moved the code to onCreate, like in the docs, and there it runs indeed very fast.
We use Koin with a lot of dependencies, so I'll look into this next week to see if I can find out what is causing this hang. Can it be that we call createTracker on a background thread and that gives issues?

**Comment by matus-tomlein** [comment]
I don't think running on a background thread should be the problem - I have just tried it and the call was instant as well.

**Comment by peterdk** [comment]
I debugged, and we indeed load our analytics module and thus also Snowplow, on a background coroutine dispatcher/thread.
I can reproduce the slow start if I make the main thread during that process busy:
```
CoroutineScope(Dispatchers.Default (  + SupervisorJob()).launch{ createTracker()}
Thread.sleep(5000)
```
What I see is that with this sleep of the main thread, the createTracker only returns after the main thread is done. Without it it returns indeed in a few 100ms. So I guess SnowPlow needs the main thread somewhere in its initialisation code and has to wait until that thread is available, before being able to return the tracker.

**Comment by peterdk** [comment]
Probably can be reproduced without coroutines, since we don't use anything here that is coroutine related.

**Comment by matus-tomlein** [comment]
Interesting! I can't think of anything in the tracker that would require running on the main thread, but will look into this!

**Comment by peterdk** [comment]
Yes, with:
```
 Thread {
            SnowPlowTracker.setup(this)
        }.start()
Thread.sleep(5000)
```

**Comment by matus-tomlein** [comment]
I think I am still missing something here, because when I tried to do this in the example app, [see here](https://github.com/matus-tomlein/snowplow-android-demo-test-lag/blob/main/app/src/main/java/com/example/testlagapp/MainActivity.kt#L42-L54), the `createTracker` call is still instant based on the log:

```
D/Snowplow: Fri Jun 23 13:45:04 GMT+02:00 2023 Snowplow tracker starting
I/TetheringManager: registerTetheringEventCallback:com.example.testlagapp
D/Snowplow: Fri Jun 23 13:45:04 GMT+02:00 2023 Snowplow tracker created
```

**Comment by peterdk** [comment]
I'll try to look in a working example there.

**Comment by peterdk** [comment]
I can't reproduce it indeed in the test project. Also not when I move it tot an Application class. I'll look into it next week to see if I can reproduce it, otherwise we'll work around it.

**Comment by peterdk** [comment]
So I profiled the boot of our app, and the Snowplow thread I use to setup the tracker waits due to the Google Advertising ID:
<img width="770" alt="Scherm­afbeelding 2023-06-23 om 14 21 42" src="https://github.com/snowplow/snowplow-android-tracker/assets/668007/8ec2a7e6-0928-4481-930d-61bafc68ceff">

**Comment by dpgmedia-gbero** [comment]
We're experiencing the same issue here. Hard to say why yet, but can you point us to where is the offloading from the main thread done in order to retrieve the `getAndroidIdfa` 

@peterdk do you have any pointer on how did you solve this ?

**Comment by dpgmedia-gbero** [comment]
Hi @matus-tomlein thanks for your reply.

I could reproduce the issue in your setup by adding gms ads dependency, see [here](https://github.com/dpgmedia-gbero/snowplow-android-demo-test-lag)

**Comment by dpgmedia-gbero** [comment]
The 10-second duration is actually caused by : 

`java.io.IOException: java.util.concurrent.TimeoutException: Timed out waiting for the service connection` in the catch of `getAndroidIdfa()`

**Comment by peterdk** [comment]
I did not yet find a proper solution for this as well. I can initialize in onCreate(), but that makes my co-devs not very happy, since we just want to do this in Koin.

I tried the 6.0.1 version, and then fetch the Idfa myself earlier in the init method, but that also caused the same blocking. 

Hopefully the repro from gbero can lead to a working fix. For now we temporarily disable idfa by returning null from PlatformContextReceiver androidIdfa method.
